src/pl/senla/lecture3/task3/building/LineStepBody.java

```diff
@@ -0,0 +1,26 @@
+package pl.senla.lecture3.task3.building;
+
+import pl.senla.lecture3.task3.part.IProductPart;
+import pl.senla.lecture3.task3.part.PartBody;
+
+import java.util.Scanner;
+
+public class LineStepBody implements ILineStep {
+
+    private IProductPart partBody;
+
+    @Override
+    public IProductPart buildProductPart() {
+        System.out.print("Input body's name -> ");
+        Scanner scanner = new Scanner(System.in);
+        String name = scanner.next();
```

Comment:
Ok. I tryed to make perfect application where there every method calls new stage of production and not only shows log.
But received finally strange (or even awfull) thing )))))
There was the most difficult thing in this task to understand life-cycle of different classes.

I return to previous commit and try again with your comments.

Reply:
Yeah, it's totally fine. You are learning new stuff and some incidents will happen. Don't let this small issues upset you :)

Just try to compare OOP to real live:
AssemblyLine should be in main class, it's your factory -> AssemplyLine should contain LineSteps (not product ;)), it's different factory conveyors  -> LineSteps should create your ProductParts -> Parts should be installed into Product with methods of IProduct.

It was almost ready with previous solution. But there Product was already created with installed Parts, because you have "All Argument Constructor". If you change it to installation parts with IProduct methods (setters) it will be exactly what you need